Date Format Correction #28

bgertonson opened the pull request:
Updated the way the SinceDate class works. It checks the kind of date, and if a kind is not specified, it assumes UTC because that is how Trello stores dates. Then when adding the date to the request, it adds it in a specific string format, instead of letting the framework decide how the date should be formatted in the URL.

bgertonson committed:
updated dates to be assumed UTC and send in proper format for predictable results

odegroot commented:
I'm voting to merge this, since this will most likely fix a bug I've run into.

See also #28.

```
TrelloNet.ITrello trello = ...;
ISince since = TrelloNet.Since.Date(dateTime);
var actions = trello.Actions.ForBoard(boardId, since: since);
```

Expected: a bunch of actions.
Actual: TrelloNet.TrelloException: invalid date
Probable reason: the date is serialized using ToString() without arguments, which is culture-dependant. I'm on a Dutch system that returns "24-6-2013 15:13:05". Trello does not recognise that format.

bgertonson committed:
Added ShortURL property to cards and LogoHash property to organizations

bgertonson committed:
Support CardCloseAction and UpdateCardPositionAction as well as test fixes

Some data has changed or expired (in the case of notifications) and tests aren't passing.
I'm not sure what the best long term fix is. But this is a common problem with integration
testing the uses real data.

bgertonson committed:
Ignored a bunch of tests that were not passing due to data that had changed or was no longer available from the Trello API and made sure that the Old data was attached to all UpdateCardAction types

bgertonson committed:
Added New ActionType : CopyCard

bgertonson committed:
Added New ActionType : CopyCard

dillenmeister commented:
I merged the first commit. The other ones are not really related to this datetime bug so please send another pull request (or several pull requests if the changes are unrelated which they seem to be a little bit) if you like.

Thanks for your contribution!
